scrunch/datasets.py

```diff
@@ -129,34 +113,23 @@ def get_dataset(dataset, connection=None, editor=False, project=None):
                 "Authenticate first with scrunch.connect() or by providing "
                 "config/environment variables")
     root = connection
-
-    if project:
-        if isinstance(project, six.string_types):
-            project_obj = get_project(project, connection)
-            ds = project_obj.get_dataset(dataset)
-        else:
-            ds = project.get_dataset(dataset)
-    else:
+    # search by dataset name
+    try:
+        shoji_ds = root.datasets.by('name')[dataset].entity
+    except KeyError:
+        # search by dataset id
         try:
-            shoji_ds = root.datasets.by('name')[dataset].entity
+            shoji_ds = root.datasets.by('id')[dataset].entity
```

Comment:
Better to factor out `dataset_catalog = root.datasets` and then do `dataset_catalog.by(...)` here and in L118, this way you only GET the datasets catalog once and not two times. If your catalog is big this will save you time.